src/main/java/pages/LoginPage.java

```diff
@@ -10,22 +11,52 @@
     private By passwordLocator = By.id("passwd");
     private By forgotPasswordLink = By.className("lost_password");
     private By signInButton = By.id("SubmitLogin");
+    private By logInErrorMessage = By.cssSelector("div.alert-danger li");
 
-    public LoginPage(WebDriver driver) {
-        super(driver);
+    public LoginPage(WebDriver driver, Logger log) {
+        super(driver, log);
     }
 
+    /** login user using Sign In button */
     public MyAccountPage logInUserSignInBtn(String email, String password) {
+        log.info("Executing LogIn test using SignIn button with email " + email + " and password " + password);
         type(email, emailLocator);
         type(password, passwordLocator);
         click(signInButton);
-        return new MyAccountPage(driver);
+        return new MyAccountPage(driver, log);
     }
 
+    /** login user using Enter key */
     public MyAccountPage logInUserEnterKey(String email, String password) {
+        log.info("Executing LogIn test using Enter Key with email " + email + " and password " + password);
         type(email, emailLocator);
         type(password, passwordLocator);
         pressKey(passwordLocator, Keys.ENTER);
-        return new MyAccountPage(driver);
+        return new MyAccountPage(driver, log);
+    }
+
+    /**login user using invalid email/ password */
+    public void negativeLogIn(String email, String password) {
+        log.info("Executing Negative LogIn test with email " + checkEmptyValue(email) + " and password " + checkEmptyValue(password));
+        type(email, emailLocator);
+        type(password, passwordLocator);
+        click(signInButton);
+    }
+
+    /**
+     * Checks if value is empty / null
+     * @param value
+     * @return "empty" if empty/null
+     */
+    public String checkEmptyValue(String value) {
```

Comment:
this method suits better into BaseSpec class; it seems to be a generic method, which can be used for values found in any Page

Reply:
Will be resolved in next PR - method was deleted as it was no longer used